homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -92,10 +176,64 @@ def finish(self):
             )
         )
 
+    async def speech_to_text(
+        self,
+        metadata: stt.SpeechMetadata,
+        stream: AsyncIterable[bytes],
+        binary_handler_id: int,
+    ) -> str:
+        """Run speech to text portion of pipeline. Returns the spoken text."""
+        engine = self.pipeline.stt_engine or "default"
+        self.event_callback(
+            PipelineEvent(
+                PipelineEventType.STT_START,
+                {
+                    "engine": engine,
+                    "metadata": asdict(metadata),
+                    "handler_id": binary_handler_id,
+                },
+            )
+        )
+
+        try:
+            # Load provider
+            stt_provider = stt.async_get_provider(self.hass, self.pipeline.stt_engine)
```

Comment:
In a future PR, should we move all these validation checks to be done before we start?